main.cpp

```diff
@@ -64,64 +64,95 @@ send me a DM to check your pull request
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
+    float value;
+    std::string name;
+    
+    T(float v, const char* nameptr)
+    {
+        value = v;
+        name = nameptr;
+    }   //1
     //2
     //3
 };
 
-struct <#structName1#>                                //4
+struct MyStruct                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }    
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float value1 { 0 }, value2 { 0 };
+    float multiply(float* updatedValue)      //12
     {
-        
+        if(updatedValue != nullptr)
+        {
+          std::cout << "U's value1 value is " << this->value1 << std::endl;
+          value1 = *updatedValue;
+          std::cout << "U's value1 new value is " << this->value1 << std::endl;
+        }
```

Comment:
if updatedValue is a nullptr, exit the function early. 
the logic that is implemented right now will cause `value 1` and `value 2` to have the same value, which will skip over the `while()` loop below.
- [ ] revise.